Makefile

```diff
@@ -390,7 +393,7 @@ kueueviz-image: kueueviz-image-build
 
 .PHONY: kueuectl
 kueuectl:
-	CGO_ENABLED=$(CGO_ENABLED) $(GO_BUILD_ENV) $(GO_CMD) build -ldflags="$(LD_FLAGS)" -o $(PROJECT_DIR)/bin/kubectl-kueue cmd/kueuectl/main.go
```

Comment:
might want to stay consistent on `$(GO_BUILD_ENV)`?

Reply:
I don't think so we need it for kueuectl. Not sure how it can be useful.

Reply:
I would suggest https://github.com/kubernetes-sigs/kueue/pull/5210#discussion_r2082154474 instead anyhow

but i think it's confusing to have GO_BUILD_ENV only apply to some targets

Reply:
alternatively, documenting which are intended to be set by callers versus internal to recursive make might be a good idea

Reply:
Let's revert this for now, as it's not part of this PR.